### PR TITLE
Add initial Prototypal-OO content.

### PR DESCRIPTION
**Pull request by chrisrhoton**

### CODE REVIEW

File: tracks/javascript/prototypal-oo/index.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Prototypal-OO
+
+Understanding prototypal inheritance (aka OLOO - Objects Linking to Other Objects) in JavaScript.
```
Comment (ericelliott) [inline tracks/javascript/prototypal-oo/index.md]: OLOO and prototypal inheritance aren't synonymous.

There are three different types of prototypal OO, of which OLOO is just one. The [Stampit docs](https://github.com/stampit-org/stampit) explain a bit. So does the ["Prototypal Inheritance with Stamps"](http://chimera.labs.oreilly.com/books/1234000000262/ch03.html#prototypal_inheritance_with_stamps) section from ["Programming JavaScript Applications"](http://pjabook.com).

**Comment by chrisrhoton** [comment]
Ah! My bad. I see the distinction you're drawing between the three types. I'll correct this, flesh it out a bit, and then resubmit!

**Comment by ericelliott** [comment]
:+1:

**Comment by ericelliott** [comment]
Don't forget ["Classical Inheritance is Obsolete"](https://vimeo.com/69255635) & [Common Misconceptions About Inheritance in JavaScript"](https://medium.com/javascript-scene/common-misconceptions-about-inheritance-in-javascript-d5d9bab29b0a),

**Comment by ericelliott** [comment]
You should probably also link to ["Prototypal Inheritance with Stamps"](http://chimera.labs.oreilly.com/books/1234000000262/ch03.html#prototypal_inheritance_with_stamps) and ["Stampit Advanced Examples"](https://github.com/stampit-org/stampit/blob/master/ADVANCED_EXAMPLES.md).

**Comment by ericelliott** [comment]
There's also [The Treaty of Orlando](http://web.media.mit.edu/~lieber/Publications/Treaty-of-Orlando-Treaty-Text.pdf) discussing differences and similarities between classes and prototypes (or "types and prototypes").

**Comment by brendt** [comment]
This book; for me, was the biggest eye opener about Prototypal OO:
https://github.com/getify/You-Dont-Know-JS/tree/master/this%20%26%20object%20prototypes

**Comment by ericelliott** [comment]
:+1:

**Comment by Webastronaut** [comment]
http://javascriptissexy.com/javascript-prototype-in-plain-detailed-language/
http://javascriptissexy.com/oop-in-javascript-what-you-need-to-know/

**Comment by ericelliott** [comment]
No to both JavaScript is Sexy links. They explain how to do objects/inheritance using the cumbersome Java look-alike hacks, leave out pros/cons of classical vs prototypal inheritance, and introduce the `new` hazard to users who won't have the experience to understand it.

**Pull request closed without merging**